### PR TITLE
Add in basic 4.0 relevant docs

### PR DESCRIPTION
**Pull request by tractorcow**
This was kind of a rough port, so @robbieaverill do you want to check if I've missed anything?

Some things that don't exist more (or yet) such as the various tasks have been removed.

### CODE REVIEW

File: docs/en/domain-configuration.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+## Deployment
+
+In order to ensure that the routing scheme will respect these domains, you should ensure the following during
+deployment:
+
+Either one of:
+
+ * Ensure that all domains configured are the only domains that the site can be accessed under, or
+ * Add `define('SS_FLUENT_FORCE_DOMAIN', true)` to your `_ss_environment.php` file, or
```
Comment (robbieaverill) [inline docs/en/domain-configuration.md]: This should be dotenv equivalent now for SS4 :)

File: docs/en/how-fluent-works.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+For example, a versioned SiteTree object will have these tables:
+
+ - SiteTree
+ - SiteTree_Live
+ - SiteTree_Versions
+ - SiteTree_Localised (joined to SiteTree)
+ - SiteTree_Localised_Live (joined to SiteTree_Live)
+ - SiteTree_Localised_Versions (joined to SiteTree_Versions)
+
+The unique key for the _Localised table is `RecordID`, `Locale`
```
Comment (robbieaverill) [inline docs/en/how-fluent-works.md]: Can we please escape the `_`_ in these examples so it doesn't break the markdown rendering

**Comment by tractorcow** [comment]
Updated @robbieaverill

File: _config/cms.yml [hunk 3]
```diff
@@ -4,6 +4,7 @@ Name: fluentcms
 SilverStripe\CMS\Model\SiteTree:
   extensions:
     FluentSiteTreeExtension: TractorCow\Fluent\Extension\FluentSiteTreeExtension
+    FluentFilteredExtension: TractorCow\Fluent\Extension\FluentFilteredExtension
```
Comment (tractorcow) [inline _config/cms.yml]: oops that's ment to be temp.

**Comment by tractorcow** [comment]
Fixed.

**Pull request merged**